contracts/src/BeefyClient.sol

```diff
@@ -321,6 +343,7 @@ contract BeefyClient {
             nextValidatorSet.id = leaf.nextAuthoritySetID;
```

Comment:
Does ` currentValidatorSet = nextValidatorSet` actually copy the counters over too? Is that confirmed in unit tests?

As that is definitely needed.

Reply:
I have refactored the counters into separate variables. This [test](https://github.com/Snowfork/snowbridge/pull/956/files#diff-2984b4351f0684d24f6eb93d006264cb1442612c4f6a9869de336506cd85f01aR25-R32) tests the overwriting of counters with new counters making sure previous values are zeroed.